packages/core/src/app/payment/Payment.tsx

```diff
@@ -377,6 +379,23 @@ class Payment extends Component<PaymentProps & WithCheckoutPaymentProps & WithLa
         }
     };
 
+    private handleError: (error: Error) => void = (error: Error) => {
+        const {
+            onUnhandledError = noop,
+            errorLogger
+        } = this.props;
+
+        const { type } = error as any;
+
+        if (type === 'unexpected_detachment') {
+            errorLogger.log(error);
+
+            return;
+        }
+        
+        return onUnhandledError(error);
```

Comment:
All look good.
Error handling normally executes in a catch block. Is it necessary to have a `return` at the end?
```
try {
       ...
        } catch (error) {
            onUnhandledError(error);
        }
```

Reply:
Thanks Peng for the feedback, here the error is already caught in SDK and passed to JS. Previously this error will be passed here and set in state of Checkout component.

Now we can handle it in child component i.e. `<Payment />` and hence the return. The rest of the errors are still passed to Checkout component.

Reply:
> Now we can handle it in child component i.e. <Payment /> and hence the return.

I understand this is the reason for line 393. Do we need a `return` on line 396 to pass an error to Checkout component?

Reply:
The return is to exit the function, otherwise the code continues and the error is passed to Checkout component. Which results in the error modal being displayed.

Reply:
> Which results in the error modal being displayed

Isn't that what we want for error types other than `unexpected_detachment`?

Reply:
Yes, the error modal is still displayed for other error types. Since we pass the execution for the error to `<Checkout />` component.